### PR TITLE
Add get, list, and get subcommands

### PR DESCRIPTION
**Pull request by eochoa260**

### CODE REVIEW

File: src/cmd/group.go [hunk 1]
```diff
@@ -38,9 +39,161 @@ EOF
...
+			group, err = getClientGQL().GetGroupWithAlias(key)
+		}
+		cobra.CheckErr(err)
+		common.PrettyPrint(group)
+		common.WasFound(group.Id == nil, key)
+	},
+}
 
-// TODO: List Groups
+var getMembersCommand = &cobra.Command{
```
Comment (rocktavious) [inline src/cmd/group.go:63]: lets call this `getGroupMembersCommand` because its nested within the group subcommand

File: src/cmd/group.go [hunk 2]
```diff
@@ -38,9 +39,161 @@ EOF
...
+		}
+		cobra.CheckErr(err)
+		members, err := group.Members(getClientGQL())
+		cobra.CheckErr(err)
+		common.PrettyPrint(members)
+		common.WasFound(group.Id == nil, key)
+	},
+}
+
+var getDescendantRepositoriesCommand = &cobra.Command{
```
Comment (rocktavious) [inline src/cmd/group.go:86]: same prefix with `getGroup...` along with all the others that are nested

File: src/cmd/group.go [hunk 3]
```diff
@@ -38,9 +39,161 @@ EOF
...
+	Use:        "repositories ID|ALIAS",
+	Short:      "Get descendant repositories for a group",
```
Comment (rocktavious) [inline src/cmd/group.go:88]: We should probably use the descriptions in the API spec

In this case specifically
```
All the repositories that fall under this group - ex. this group's child repositories, all the child repositories of this group's descendants, etc.
```
Maybe just for the Long description and then for the short lets do something like
```
Get all repositories for a group
```

File: src/cmd/group.go [hunk 4]
```diff
@@ -38,9 +39,161 @@ EOF
...
+	Long:       `Get descendant repositories for a group`,
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID", "ALIAS"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		var group *opslevel.Group
+		var err error
+		if common.IsID(key) {
```
Comment (rocktavious) [inline src/cmd/group.go:96]: Since this is repeated more then 3 times it should be pull out to a reusable func

File: src/cmd/group.go [hunk 5]
```diff
@@ -38,9 +39,161 @@ EOF
...
+			group, err = getClientGQL().GetGroup(key)
+		} else {
+			group, err = getClientGQL().GetGroupWithAlias(key)
+		}
+		cobra.CheckErr(err)
+		common.PrettyPrint(group)
+		common.WasFound(group.Id == nil, key)
```
Comment (rocktavious) [inline src/cmd/group.go:59]: this `WasFound` method needs to come before the `PrettyPrint` so the error is thrown first

**Review by rocktavious — CHANGES_REQUESTED** [review]
Just a few things to fix up and some redundant code that can be DRY'd up.

File: src/cmd/group.go (hunk 5 above)
Comment (eochoa260) [reply src/cmd/group.go:59]: I thought the same thing when I checked to see what it did. I'll move it to before and check where I got this from to see if the same needs to be done elsewhere.

File: src/cmd/group.go (hunk 4 above)
Comment (eochoa260) [reply src/cmd/group.go:96]: Yeah, and I got it from somewhere else so I'm sure it's the same case for other get commands

File: src/cmd/group.go (hunk 2 above)
Comment (rocktavious) [reply src/cmd/group.go:86]: IMO more explicit is always better unless its customer facing.

**Review by rocktavious — COMMENTED** [review]
LGTM

**Review by rocktavious — APPROVED** [review]

**Pull request merged**